src/DependencyLoader.php

```diff
@@ -19,23 +21,36 @@ public function __construct(DependencyParser $dependencyParser)
         $this->dependencyParser = $dependencyParser;
     }
 
+    /**
+     * @throws CommandExecutionException
+     */
     public function loadDependencies(string $composer, string $project): array
     {
         $commandOutput = $this->runComposerLicenseCommand($composer, $project);
 
-        return $this->dependencyParser->parse(join(PHP_EOL, $commandOutput));
+        return $this->dependencyParser->parse(implode("\n", $commandOutput));
     }
 
-    private function runComposerLicenseCommand(string $composer, string $project, string $format = 'json'): array
+    /**
+     * @throws CommandExecutionException
+     */
+    private function runComposerLicenseCommand(string $composer, string $project): array
     {
-        $command = sprintf('%s licenses -f %s -d %s', $composer, $format, $project);
+        $command = sprintf('%s licenses --format json --working-dir %s', escapeshellarg($composer), escapeshellarg($project));
 
         return $this->exec($command);
     }
 
-    protected function exec(string $command)
+    /**
+     * @throws CommandExecutionException
+     */
+    protected function exec(string $command): array
     {
-        exec($command, $output);
+        exec($command, $output, $exitCode);
+
+        if ($exitCode !== 0) {
```

Comment:
```suggestion
        if ($exitCode !== Command::SUCCESS) {
```

Reply:
I wouldn't change the zero. The exit code coming from Composer (or the attempted shell command in general) supposedly comes from the documentation (see https://getcomposer.org/doc/03-cli.md#process-exit-codes), and zero is the obvious successful result. The constant `Command::SUCCESS` has the same numeric value, but I understand it to be used as a named exit code for Symfony console programs.

So mixing explicit integers when calling a distinct composer command with the potential return values of this program sounds odd. At least I wouldn't add more error cases for Composer returning 1 or 2 as an exit code with `Command::FAILURE` for "Generic/unknown error code" and `Command::INVALID` for "Dependency solving error code".